kvs-gbif/src/main/java/org/gbif/kvs/species/BackboneMatchByID.java

```diff
@@ -0,0 +1,165 @@
+        if (checklistKey.equalsIgnoreCase(NUB_DATASET_KEY.toString())) {
+          try {
+            backboneUsageKey = Integer.parseInt(adjustedID);
+          } catch (NumberFormatException e) {
+            issues.add(notFoundFlag);
+            LOG.debug("Adjusted ID[{}] is no backbone key integer", adjustedID);
+          }
+
+        } else {
+          NameUsageSearchResponse response = checklistbankService.lookupNameUsage(checklistKey, adjustedID);
+
+          if (response.getResults().isEmpty()) {
+            issues.add(notFoundFlag);
+            LOG.debug("No matches found when looking up [{}] against checklist[{}]", adjustedID, checklistKey);
+
+          } else if (response.getResults().size()==1) {
+            NameUsageSearchResponse.Result match = response.getResults().get(0);
+            LOG.debug("Looking up [{}] against checklist[{}] matched to scientificName[{}], nameUsageID[{}], backboneID[{}]", adjustedID, checklistKey, match.getScientificName(), match.getKey(), match.getNubKey());
+            if (match.getNubKey() != null) {
+              compareAndFlagDifferingNames(id, name, issues, match);
+              backboneUsageKey = match.getNubKey();
+
+            } else {
+              issues.add(ignoredFlag); // concept found, but ignored as it doesn't snap to the backbone
+            }
+
+          } else {
+            issues.add(ignoredFlag); // we did not find a single concept to use (an edge case indicating bad checklist data)
+            LOG.debug("{} matches found looking up [{}] against checklist[{}]. Ignored as ambiguous", response.getResults().size(), adjustedID, checklistKey);
+          }
+        }
+      } else {
+        issues.add(ignoredFlag);
+        LOG.debug("Adjusted ID[{}] is not configured to match against any checklist by it's prefix", adjustedID);
+      }
+    }
+    return backboneUsageKey;
+  }
+
+  /**
+   * Add a flag only if the provided name and match response can be parsed and the canonical names found to differ.
+   * This lenient in behaviour as it is only a hint to publishers so should not fail.
+   */
+  static void compareAndFlagDifferingNames(String id, String name, Set<OccurrenceIssue> issues, NameUsageSearchResponse.Result match) {
+    try {
+      ParsedName recordName = PARSER.parse(name, null, null);
+      ParsedName foundName = PARSER.parse(match.getScientificName(), null, null);
+      boolean equal = recordName!=null && foundName!=null &&
+          StringUtils.equals(recordName.canonicalName(), foundName.canonicalName());
+      if (!equal) {
+        LOG.debug("Canonical name[{}] found using ID[{}] does not match canonical name[{}] on record, after parsing", foundName.canonicalName(), id, recordName.canonicalName());
+        issues.add(SCIENTIFIC_NAME_AND_ID_INCONSISTENT); // we will still use the found concept though
+      }
+    } catch(Exception e) {
```

Comment:
Maybe catch UnparsableNameException explicitly as these are rather likely to occur and handle only as warning?

Reply:
Will do